src/redisraft.c

```diff
@@ -549,11 +549,60 @@ static bool handleInterceptedCommands(RedisRaftCtx *rr,
     return false;
 }
 
+static bool AskingHandleCommand(RedisRaftCtx *rr, RedisModuleCtx *ctx, RaftRedisCommandArray *cmds)
+{
+    if (cmds->len != 1) {
+        return false;
+    }
+
+    /* TODO: will have to be == 2 in future with magic value in asking for argv[2] */
+    if (cmds->commands[0]->argc != 1) {
+        return false;
+    }
+
+    /* Is this a ASKING command? */
+    RaftRedisCommand *cmd = cmds->commands[0];
+    size_t cmd_len;
+    const char *cmd_str = RedisModule_StringPtrLen(cmd->argv[0], &cmd_len);
+
+    if (cmd_len != 6 || strncasecmp(cmd_str, "ASKING", 6) != 0) {
+        return false;
+    }
+
+    Node *ret;
+    if (checkLeader(rr, ctx, &ret) == RR_ERROR) {
+        /* change to replyAsk once that's committed */
+        replyRedirect(ctx, -1, &ret->addr);
+        return true;
+    }
+
+    ClientState *clientState = ClientStateGet(rr, ctx);
+    clientState->asking = true;
+
+    RedisModule_ReplyWithSimpleString(ctx, "OK");
+
+    return true;
+}
+
+static bool GetAskingState(RedisRaftCtx *rr, RedisModuleCtx *ctx)
+{
+    ClientState *clientState = ClientStateGet(rr, ctx);
+    return clientState->asking;
+}
+
 static void handleRedisCommand(RedisRaftCtx *rr,
                                RedisModuleCtx *ctx, RaftRedisCommandArray *cmds)
 {
     Node *leader_proxy = NULL;
 
+    if (AskingHandleCommand(rr, ctx, cmds)) {
```

Comment:
```suggestion
    if (handleAsking(rr, ctx, cmds)) {
```

Style. Local functions start with lowercase letter.